miden-lib/comment_format.md

```diff
@@ -0,0 +1,224 @@
+
+### Inputs
+
+Inputs block could contain three components: operand stack, advice stack and advice map. Description of the each container should be offseted with two spaces relative to the start of the `Inputs` word. Each name of the container should be separated from its value by the colon (e.g. `Operand stack: [value_1]`).
+
+Operand stack and advice stack should be presented as an array containing some data.
+
+The lines which exceed 100 symbols should be formatted differently, it could be done in two different ways:
+
+1. The line should be broken, and the end of the line should be moved to the new line with an offset such that the first symbol of the first element on the second line should be directly above the first symbol of the first element on the first line (see the value of the `FOREIGN_ACCOUNT_ID` in the example in `Formats` section).
+2. The exceeded array should be formatted in a column, forming a Word or some other number of related elements on each line. Each new line should be offseted with two spaces relative to the name of the container (see example below).
+
+Example:
+
+```masm
+#! Inputs:
+#!   Operand stack: []
+#!   Advice stack: [
+#!     account_id, 0, 0, account_nonce, 
+#!     ACCOUNT_VAULT_ROOT, 
+#!     ACCOUNT_STORAGE_COMMITMENT, 
+#!     ACCOUNT_CODE_COMMITMENT
+#!   ]
+```
+
+To show that some internal value array could have dynamic length, additional brackets should be used (see the `[VALUE_B]` in the advice stack in the example in `Formats` section).
+
+In case some inputs are presented on the stack only if some condition is satisfied, such inputs should be placed in the "optional" box: inside the parentheses with a question mark at the end. Opening and closing brackets should be placed on a new line with the same offset as the other inputs, and values inside the brackets should be offseted by two spaces.
+
+Example:
+
+```masm
+#!   ...
+#!   Advice stack: [
+#!      NOTE_METADATA,
+#!      assets_count,
+#!      (
+#!        block_num,
+#!        BLOCK_SUB_HASH,
+#!        NOTE_ROOT,
+#!      )?
+#!   ]
+#!   ...
+```
+
+Advice map should be presented as a sequence of the key-value pairs in the curly brackets. Opening bracket should stay on the same line, and the closing bracket should be placed on the next line after the last key-value pair with the same offset as the `Advice map` phrase.
+
+Each pair should start at the new line with additional two spaces offset relative to the start of the `Advice map` phrase. Pairs should be separated with comma. The same formatting rules as to the operand and advice stacks should be applied for the each key-value pair.
+
+### Outputs
+
+Outputs should show the final state of each container, used in the inputs, except for the advice map: almost always the final state of the advice map is unimportant, since it is always the same as at the beginning of the procedure execution (see the example in the `Formats` section below).
```

Comment:
> almost always the final state of the advice map is unimportant, since it is always the same as at the beginning of the procedure execution

Nit: It's not entirely clear to me whether the advice map is _always_ the same or _almost always_. Maybe this sentence could clarify.

Reply:
We have some instructions which put some data to the advice map, and I thought that we never used them in the `miden-base`, but I was wrong. I'll rewrite this paragraph and update the code which uses these instructions.